test/e2e/play_kube_test.go

```diff
@@ -1098,6 +1115,52 @@ var _ = Describe("Podman play kube", func() {
 		Expect(label).To(ContainSubstring("unconfined_u:system_r:spc_t:s0"))
 	})
 
+	It("podman play kube should use default infra_image", func() {
+		err := writeYaml(checkInfraImagePodYaml, kubeYaml)
+		Expect(err).To(BeNil())
+
+		kube := podmanTest.Podman([]string{"play", "kube", kubeYaml})
+		kube.WaitWithDefaultTimeout()
+		Expect(kube).Should(Exit(0))
+
+		podInspect := podmanTest.Podman([]string{"inspect", "check-infra-image", "--format", "'{{ .InfraContainerID }}'"})
+		podInspect.WaitWithDefaultTimeout()
+		infraContainerID := podInspect.OutputToString()
+
+		conInspect := podmanTest.Podman([]string{"inspect", infraContainerID, "--format", "'{{ .ImageName }}'"})
+		conInspect.WaitWithDefaultTimeout()
+		infraContainerImage := conInspect.OutputToString()
+		Expect(infraContainerImage).To(Equal(config.DefaultInfraImage))
+	})
+
+	It("podman play kube should use customized infra_image", func() {
+		conffile := filepath.Join(podmanTest.TempDir, "container.conf")
+
+		infraImage := "k8s.gcr.io/pause:3.2"
+		err := ioutil.WriteFile(conffile, []byte(fmt.Sprintf("[engine]\ninfra_image=\"%s\"\n", infraImage)), 0644)
+		Expect(err).To(BeNil())
+
+		os.Setenv("CONTAINERS_CONF", conffile)
+
```

Comment:
You need to use `defer os.Unsetenv("CONTAINERS_CONF")`

Reply:
I updated the code to use defer to unset the env, but seems the env is not properly set.

In this test, the play kube does not use the customized infra image.

The code change in file ` pkg/domain/infra/abi/play.go` works well, I used it for several days.

Do you have any ideas on this?

https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6372351924240384/html/int-remote-ubuntu-2010-root-host.log.html#t--podman-play-kube-should-use-customized-infra_image--1

Reply:
The server does not pick up config changes after it is started. Use this:
https://github.com/containers/podman/blob/8a55363b80a012b6c6e774957ac477d0fd540a5e/test/e2e/containers_conf_test.go#L200-L203